src/robotide/postinstall/__main__.py

```diff
@@ -211,29 +211,28 @@ def _create_desktop_shortcut_linux(frame=None):
 def _create_desktop_shortcut_mac(frame=None):
     import os
     import shutil
-    import subprocess
     ride_app_name = 'RIDE.app'
     application_path = '/Applications'
     ride_app_pc_path = os.path.join(application_path, ride_app_name)
     ride_app_module_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), ride_app_name)
-    
-    if not option_q and not option_f and not _askyesno("Setup", "Create application shortcut?", frame):
-        return False
-    app_script = os.path.join(ride_app_module_path, 'Contents', 'MacOS', 'RIDE')
-    with open(app_script, 'w+') as shortcut:
-        shortcut.write("#!/bin/sh\nPAL=$PATH\nfor i in `cat /etc/paths`\n    do\n        PAL=\"$PAL:$i\"\n"
-                       "    done\nPATH=$PAL\nexport $PATH\n{} -m robotide.__init__ $* 2>"
-                       " /dev/null &\n".format(sys.executable))
-    if not exists(ride_app_pc_path):
-        shutil.copytree(ride_app_module_path, ride_app_pc_path)
-    elif option_f:
-        shutil.rmtree(ride_app_pc_path, True)
+
+    if not exists(ride_app_pc_path) or option_f:
+        if not option_q and not option_f and not _askyesno("Setup", "Create application shortcut?", frame):
+            return False
+        app_script = os.path.join(ride_app_module_path, 'Contents', 'MacOS', 'RIDE')
+        with open(app_script, 'w+') as shortcut:
+            shortcut.write("#!/bin/sh\nPAL=$PATH\nfor i in `cat /etc/paths`\n    do\n        PAL=\"$PAL:$i\"\n"
+                           "    done\nPATH=$PAL\nexport $PATH\n{} -m robotide.__init__ $* 2>"
+                           " /dev/null &\n".format(sys.executable))
+        if exists(ride_app_pc_path):
+            shutil.rmtree(ride_app_pc_path, True)
         shutil.copytree(ride_app_module_path, ride_app_pc_path)
-    user = str(subprocess.check_output(['logname']).strip(), encoding='utf-8')
-    userdesktoplink = '/Users/' + user + '/Desktop/' + ride_app_name
-    if not exists(userdesktoplink) or option_f:
+        user_desktop_link = os.path.join(os.path.expanduser('~'), 'Desktop', ride_app_name)
```

Comment:
@jnhyperion This code should not be changed, This is the second time you do this.
I have coded this way because when the user is installing with `sudo -H` my code returns the username. With your code it will return `root`.

Reply:
@HelioGuilherme66 it's because in my Mac the result is contrary to your results:

`
print(str(subprocess.check_output(['logname']).strip(), encoding='utf-8'))
print(os.path.expanduser('~'))

>> root
>> /Users/jnhuang
`

Reply:
In my PyCharm terminal: `logname -> root`
In my system termimal: `logname -> jnhuang`

I think we should not use system command to get user desktop path, because it has system configuration dependencies, while os.path module not.

If user use 'sudo -H' install RIDE, which means user intent to install RIDE under user root.

Reply:
I think if the user uses `sudo -H` it means install for all users, for example if is an IT people installing remotely in the network.